File: 2718-sum-of-matrix-after-queries/2718-sum-of-matrix-after-queries.cpp
class Solution {
public:
    long long matrixSumQueries(int n, vector<vector<int>>& queries) {
        
        long long ans=0;
        vector<bool>rows(n,true),cols(n,true);
        int col=n,row=n;
        for(int i=queries.size()-1;i>=0;i--){
            if(queries[i][0]==0 && rows[queries[i][1]]){
                rows[queries[i][1]]=false;
                ans+=queries[i][2]*col;
                row--;
            }
            else if(queries[i][0]==1 && cols[queries[i][1]]){
                cols[queries[i][1]]=false;
                ans+=queries[i][2]*row;
                col--;
            }
        }
        return ans;
        
    }
};

File: 2718-sum-of-matrix-after-queries/2718-sum-of-matrix-after-queries.java
class Solution {
    public long matrixSumQueries(int n, int[][] q) {
     
        long ans=0;
        boolean []cols=new boolean[n];
        boolean []rows=new boolean[n];
        
        Arrays.fill(cols,true);
        Arrays.fill(rows,true);
    
        int row=n,col=n;
        
        for(int i=q.length-1;i>=0;i--){
            if(q[i][0]==0 && rows[q[i][1]]==true)
            {
                rows[q[i][1]]=false;
                ans+=q[i][2]*col;
                row--;
            }
            else if(q[i][0]==1 && cols[q[i][1]]==true)
            {
                cols[q[i][1]]=false;
                ans+=q[i][2]*row;
                col--;
            }
        }
    
        return ans;
    
    }
}